Tasks/Gradle/CodeAnalysis/Common/BuildOutput.ts

```diff
@@ -0,0 +1,70 @@
+import {ModuleOutput} from './ModuleOutput';
+
+import path = require('path');
+import fs = require('fs');
+import glob = require('glob');
+
+import tl = require('vsts-task-lib/task');
+
+
+export enum BuildEngine {
+    Maven, 
+    Gradle
+}
+
+/**
+ * Build output from a single or multi module project. Identifies modules based on path conventions. 
+ * 
+ * @export
+ * @class BuildOutput
+ * @implements {IBuildOutput}
+ */
+export class BuildOutput {
```

Comment:
The differences between Maven and Gradle are minimal in terms of where the build artifacts get stored so there's no reason to complicate things and introduce specific classes.